Show progress dialog during startup

chrdavis opened the pull request:
This change adds a standard shell progress dialog during startup to give feedback to the user that we are gathering all the items that were selected (including items in subfolders).  This allows the user to know we are still doing work as well as provide a means to cancel the operation.

![image](https://user-images.githubusercontent.com/6734176/105622807-6a68e100-5dc9-11eb-9480-2bd74ff9ae99.png)

Applies to #9254

niels9001 commented:
Minor nit: to be consistent, can we use the same .ico that we use in the Shell right-click context menu in the dialog topbar?

htcfreek commented:
@chrdavis 
Can you chage the word "fixes" to "Applies to". We don't use the wording "fixes" to not auto close the issue.

chrdavis commented:
> @chrdavis
> Can you chage the word "fixes" to "Applies to". We don't use the wording "fixes" to not auto close the issue.

Done

chrdavis commented:
Any update on this review?

enricogior commented:
@chrdavis 
is it expected that the progress dialog stays open for about 1 second after the PowerRename dialog opens?

![rename](https://user-images.githubusercontent.com/3206696/106261303-afee3980-6221-11eb-8fe8-815b505c6266.gif)

chrdavis commented:
> @chrdavis
> is it expected that the progress dialog stays open for about 1 second after the PowerRename dialog opens?
> 
> ![rename](https://user-images.githubusercontent.com/3206696/106261303-afee3980-6221-11eb-8fe8-815b505c6266.gif)

The progress dialog should show only after a delay of a few seconds.  Perhaps I can wrap this in a better delay of my own on top of that.

chrdavis commented:
I updated SmartRename with a progress dialog that takes into account ~ 3s before being shown.  I'll let that get downloaded and used then port it here.

enricogior commented:
@chrdavis 
what should we do with the current PR?

chrdavis commented:
> @chrdavis
> what should we do with the current PR?

I ported the changes.  Please review when you get a chance.

enricogior commented:
@chrdavis 
I tested the updated version and it looks like the progress bar is slowing down the file processing.
I used the repository folders so it's easy to replicate.
When renaming individual folders, it take in total 3-4 seconds.
But when renaming the folders together, the progress bar shows up and it takes over 10 seconds.
The overhead introduced by the progress bar keep increasing for larger folders.
I would guess that the problem is caused by updating the progress bar with the filenames, can the progress bar just show the bar and not the filenames?
Also, it takes about 3 seconds for the progress bar to show, what about lowering to 2 seconds or even 1.5 seconds?

![image](https://user-images.githubusercontent.com/3206696/108190899-cbf34580-7112-11eb-96a8-f5e0b2a668cc.png)

chrdavis commented:
Updated based on the feedback.  The updating of the progress dialog is replaced with a static message now so the slowdown should go away.  I also changed the delay from 3s to 1.5s.

crutkas commented:
@enricogior any reason why this can't be in?

enricogior commented:
@crutkas 
it is still not working for me, after the last change, the progress dialog appears after the PowerRename dialog is shown.

chrdavis commented:
> @crutkas
> it is still not working for me, after the last change, the progress dialog appears after the PowerRename dialog is shown.

Really?  Can you post a gif?

chrdavis commented:
I just updated the timeout from 1500ms to 2500ms.  Originally it was 3000ms.

enricogior commented:
@chrdavis 
this was before your last commit:

![rename](https://user-images.githubusercontent.com/3206696/109276128-ecf41e80-7815-11eb-97d9-35195f0fe3e4.gif)

I'll rebuild it and test it again

enricogior commented:
@chrdavis 
same behavior using the latest bits:

![rename2](https://user-images.githubusercontent.com/3206696/109276713-a4893080-7816-11eb-81d2-06751d80ed99.gif)

chrdavis commented:
Yeah, we are competing on the UI thread.  I'll need to shuffle some things around.  I'll have an update next week.

chrdavis commented:
Updated once again.  Progress UI is now invoked on a background thread.  No longer need the delay timer for the progress dialog.  Please try this out.

chrdavis commented:
@crutkas  Is this tested and good to go for next release?

crutkas commented:
@enricogior assuming won't happen for 35 due to time. What needs to happen to review here.

enricogior commented:
@crutkas @chrdavis 
I'll review it after 0.35 is released.

enricogior commented:
@chrdavis 
tested, it works as expected, thanks for making it perfectly smooth!
